### PR TITLE
Replace ChannelOption if / else checks with Map lookups

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Finding the right method to call for a specific ChannelOption may need multiple if branches.

Modifications:

Use a Map that holds mappings for fast lookup

Result:

Cleaner code and less if / else

### CODE REVIEW

**Comment by normanmaurer** [comment]
@franz1981 @trustin @chrisvest @Scottmitch I only implemented this for AbstractChannel and NioDatagramChannel atm to show the idea. I am not sure if this is better then what we have atm

File: transport/src/main/java/io/netty5/channel/AbstractChannel.java [hunk 1]
```diff
@@ -1412,11 +1367,36 @@ protected boolean isExtendedOptionSupported(ChannelOption<?> option) {
         return false;
     }
 
-    private static Set<ChannelOption<?>> supportedOptions() {
-        return newSupportedIdentityOptionsSet(
-                AUTO_READ, WRITE_BUFFER_WATER_MARK, CONNECT_TIMEOUT_MILLIS, MAX_MESSAGES_PER_READ,
-                WRITE_SPIN_COUNT, BUFFER_ALLOCATOR, RCVBUF_ALLOCATOR, AUTO_CLOSE, MESSAGE_SIZE_ESTIMATOR,
-                MAX_MESSAGES_PER_WRITE, ALLOW_HALF_CLOSURE);
+    private static Map<ChannelOption<?>, ChannelOptionMapping<AbstractChannel<?, ?, ?>, ?>> supportedMappings() {
+        Map<ChannelOption<?>, ChannelOptionMapping<AbstractChannel<?, ?, ?>, ?>> mappingMap = new IdentityHashMap<>();
```
Comment (hyperxpro) [inline transport/src/main/java/io/netty5/channel/AbstractChannel.java:1371]: Should we define capacity while initializing?

**Comment by chrisvest** [comment]
I don't think map lookups can constant fold.

We could reduce it to a single if-else chain by having a protected method that return VarHandles to the field-based options, which sub-classes can override and add to. Then getOption/setOption/isOptionSupported would rely on that, in addition to having their own short if-else chains for the occasional non-field-based option.

I suggest trying the different approaches in a benchmark and see how they fare.

**Comment by normanmaurer** [comment]
> I don't think map lookups can constant fold.
> 
> 
> 
> We could reduce it to a single if-else chain by having a protected method that return VarHandles to the field-based options, which sub-classes can override and add to. Then getOption/setOption/isOptionSupported would rely on that, in addition to having their own short if-else chains for the occasional non-field-based option.
> 
> 
> 
> I suggest trying the different approaches in a benchmark and see how they fare.

Imho this looks very ugly for a method that users might want to implement / override

**Comment by normanmaurer** [comment]
Let close this for now

**Pull request closed without merging**